pkg/api/utils/eventwatch.go

```diff
@@ -0,0 +1,134 @@
+package api
+
+import (
+	"context"
+	"github.com/keptn/go-utils/pkg/api/models"
+	"log"
+	"sort"
+	"time"
+)
+
+// EventWatcher implements the logic to query for events and provide them to the client
+type EventWatcher struct {
+	nextCEFetchTime time.Time
+	eventHandler    EventHandlerInterface
+	eventFilter     EventFilter
+	ticker          *time.Ticker
+	timeout         <-chan time.Time
+	manipulator     EventManipulatorFunc
+}
+
+// Watch starts the watch loop and returns a channel to get the actual events as well as a context.CancelFunc in order
+// to stop the watch routine
+func (ew *EventWatcher) Watch(ctx context.Context) (<-chan []*models.KeptnContextExtendedCE, context.CancelFunc) {
+	ctx, cancel := context.WithCancel(ctx)
+	ch := make(chan []*models.KeptnContextExtendedCE)
+	go ew.fetch(ctx, cancel, ch, ew.eventFilter)
+	return ch, cancel
+}
```

Comment:
This sounds like an important utility function for devs :+1:

Can you please add a usage example in our [README#usage](https://github.com/keptn/go-utils#usage)?

Reply:
Done, I've also took the opportunity to cleanup the README file in general.